### PR TITLE
Add tinyexr 20200811

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **tinyexr/20200811**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by danimtb — DISMISSED** [review]

File: recipes/tinyexr/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "cci.20200811":
+    url: "https://github.com/syoyo/tinyexr/archive/cf8550f1b8b9f5f79f02df810c885ed2a2b578f9.tar.gz"
```
Comment (uilianries) [inline recipes/tinyexr/all/conandata.yml:3]: There are official releases and project seems be still active. The last release is from 2016. Please, ask for a new tag to the upstream

**Review by uilianries — CHANGES_REQUESTED** [review]

File: recipes/tinyexr/all/conandata.yml (hunk 1 above)
Comment (intelligide) [reply recipes/tinyexr/all/conandata.yml:3]: Master branch can be considered stable https://github.com/syoyo/tinyexr/issues/149#issuecomment-689336326

Comment (Croydon) [reply recipes/tinyexr/all/conandata.yml:3]: Now we got a `1.0.0` release 😄

Comment (uilianries) [reply recipes/tinyexr/all/conandata.yml:3]: Thank you people for asking to the author!

**Review by uilianries — DISMISSED** [review]

File: recipes/tinyexr/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        content_lines = open(os.path.join(self.source_folder, self._source_subfolder, "tinyexr.h")).readlines()
+        license_content = []
+        for i in range(1, 25):
+            license_content.append(content_lines[i][:-1])
+        tools.save("LICENSE", "\n".join(license_content))
+
+    def package(self):
+        self.copy("tinyexr.h", dst="include", src=self._source_subfolder)
+        self._extract_license()
+        self.copy("LICENSE", dst="licenses")
```
Comment (madebr) [inline recipes/tinyexr/all/conanfile.py:58]:
```suggestion
    @property
    def _extract_license(self):
        content_lines = open(os.path.join(self.source_folder, self._source_subfolder, "tinyexr.h")).readlines()
        license_content = []
        for i in range(1, 25):
            license_content.append(content_lines[i][:-1])
        return license_content

    def package(self):
        tools.save(os.path.join(self.package_folder, "LICENSE") self._extract_license)
        self.copy("LICENSE", dst="licenses")
```

File: recipes/tinyexr/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+class TinyExrConan(ConanFile):
+    name = "tinyexr"
+    description = "Tiny OpenEXR image loader/saver library"
+    homepage = "https://github.com/syoyo/tinyexr"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    topics = ("conan", "exr", "header-only")
+    options = {
+        "with_miniz": [True, False],
```
Comment (madebr) [inline recipes/tinyexr/all/conanfile.py:14]:
```suggestion
        "with_zlib": [True, False],
```

File: recipes/tinyexr/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        "with_piz": [True, False],
+        "with_zfp": [True, False],
+        "with_thread": [True, False],
+        "with_openmp": [True, False],
+    }
+    default_options = {
+        "with_miniz": True,
```
Comment (madebr) [inline recipes/tinyexr/all/conanfile.py:21]:
```suggestion
        "with_zlib": True,
```

File: recipes/tinyexr/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        "with_openmp": False,
+    }
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if not self.options.with_miniz:
```
Comment (madebr) [inline recipes/tinyexr/all/conanfile.py:34]:
```suggestion
        if not self.options.with_zlib:
```

File: recipes/tinyexr/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    def package(self):
+        self.copy("tinyexr.h", dst="include", src=self._source_subfolder)
+        self._extract_license()
+        self.copy("LICENSE", dst="licenses")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.defines.append("TINYEXR_USE_MINIZ=" + ( "1" if self.options.with_miniz else "0"))
```
Comment (madebr) [inline recipes/tinyexr/all/conanfile.py:64]:
```suggestion
        self.cpp_info.defines.append("TINYEXR_USE_MINIZ=" + ( "1" if self.options.with_zlib else "0"))
```

File: recipes/tinyexr/all/conanfile.py (hunk 3 above)
Comment (intelligide) [reply recipes/tinyexr/all/conanfile.py:14]: It uses miniz, not minizip.

Comment (madebr) [reply recipes/tinyexr/all/conanfile.py:14]: The option is called miniz, but you still require conan's zlib.

Comment (intelligide) [reply recipes/tinyexr/all/conanfile.py:14]: zlib is only used when miniz is disabled.
I think we must respect the options imposed by the author of the library (https://github.com/syoyo/tinyexr#compile-flags). I personally think it is better to use `use_miniz` for `TINYEXR_USE_MINIZ`. With `use_zlib`, the consumer might wonder why there is no option for miniz after reading the documentation.

Comment (madebr) [reply recipes/tinyexr/all/conanfile.py:14]: Ok, I understand.
Then I see the following problem: you haven't added `miniz` as a requirement.
Also, maybe add the following option: `"with_z": ["zlib", "miniz"]` to make it clear to consumers that they have the choice between miniz and zlib.

Comment (intelligide) [reply recipes/tinyexr/all/conanfile.py:14]: miniz is embedded. I can remove it but but it would be a huge patch (~6459 continuous lines)

Comment (madebr) [reply recipes/tinyexr/all/conanfile.py:14]: Is it a separate header file or embedded in the main header?
If it's embedded, can you surround the miniz code with `#if 0` and `#endif`?`

Comment (intelligide) [reply recipes/tinyexr/all/conanfile.py:14]: It's embedded in the main header (See https://github.com/syoyo/tinyexr/blob/master/tinyexr.h#L591). Surrounding it with macros may work.

Comment (intelligide) [reply recipes/tinyexr/all/conanfile.py]: The patch is a bit heavier than expected but it works.

Comment (madebr) [reply recipes/tinyexr/all/conanfile.py]: Great!
Now we will both sleep better 🤣

File: recipes/tinyexr/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        if self.options.with_z == "miniz":
+            self.requires("miniz/2.1.0")
+        else:
+            self.requires("zlib/1.2.11")
+        if self.options.with_zfp:
+            self.requires("zfp/0.5.5")
+
+    def configure(self):
+        if self.options.with_thread:
+            tools.check_min_cppstd(self, "11")
```
Comment (prince-chrismc) [inline recipes/tinyexr/all/conanfile.py:44]:
```suggestion
        if self.options.with_thread and self.settings.compiler.get_safe("cppstd") :
            tools.check_min_cppstd(self, "11")
```

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

File: recipes/tinyexr/all/patches/0001-use-conan-deps.patch [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+ 
+ // static bool IsBigEndian(void) {
+ //  union {
+@@ -7650,11 +7653,11 @@
+   // Compress the data using miniz
+   //
+ 
+-  miniz::mz_ulong outSize = miniz::mz_compressBound(src_size);
+-  int ret = miniz::mz_compress(
++  mz_ulong outSize = mz_compressBound(src_size);
```
Comment (SSE4) [inline recipes/tinyexr/all/patches/0001-use-conan-deps.patch:38]: why namespace was changed?

Comment (intelligide) [reply recipes/tinyexr/all/patches/0001-use-conan-deps.patch:38]: Embedded miniz was surrounded with `miniz` namespace and original miniz doesn't use any namespace. I changed tinyexr to use miniz from CCI.

**Pull request merged**